sdk-cpp/src/internal/do_download_property_internal.h

```diff
@@ -0,0 +1,71 @@
+#ifndef _DELIVERY_OPTIMIZATION_DO_DOWNLOAD_PROPERTY_INTERNAL_H
+#define _DELIVERY_OPTIMIZATION_DO_DOWNLOAD_PROPERTY_INTERNAL_H
+
+#if defined(DO_INTERFACE_COM)
+#include <OAIdl.h>
+#else
+#include <boost/variant.hpp>
+#endif
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "do_download_property.h"
+
+namespace microsoft
+{
+namespace deliveryoptimization
+{
+namespace details
+{
+class CDownloadPropertyValueInternal
```

Comment:
re:
"We don't need a separate internal class when the native_type is controlled by a compile-time constant. Or am I missing something?"

I'm not sure if there's a way to do the above. Need some way to hide the platform specific implementation. 

Considered forward declaring the typdef, but it also requires the type to be fwd declared.

Reply:
One thing I had considered was just removing the do_download_property class altogether, and just providing overloads for msdo::download::set_property() to accept all the different types, and then to convert the type to the respective native type in the download_impl class.
I forget why we wanted to let the caller manage a download_property class, because once it's constructed it's immutable in the current implementation.

Reply:
Providing overloads for msdo::download::set_property() --
sounds good. See how it works out in terms of coding and API clarity.

But what about get_property? The caller will not know which overload to call.

Reply:
I agree the pimpl idiom is not great just so we can abstract away the underlying native type that we use to make calls to our client, but I'm not too sure on which I prefer between what's in this PR and then using multiple overloads for set_property()... thinking